tests/system/Helpers/ArrayHelperTest.php

```diff
@@ -219,8 +219,17 @@ public function testArraySortByMultipleKeysFailsEmptyParameter($data, $sortColum
 	 */
 	public function testArraySortByMultipleKeysFailsInconsistentArraySizes($data)
 	{
-		$this->expectException('ErrorException');
-		$this->expectExceptionMessage('Array sizes are inconsistent');
+		// PHP 8 changes this error type
+		if (version_compare(PHP_VERSION, '8.0', '<'))
+		{
+			$this->expectException('ErrorException');
+			$this->expectExceptionMessage('Array sizes are inconsistent');
+		}
+		else
+		{
+			$this->expectException('ValueError');		
+			$this->expectExceptionMessage('Array sizes are inconsistent');
+		}
```

Comment:
```suggestion
			$this->expectException('ErrorException');
		}
		else
		{
			$this->expectException('ValueError');
		}
		
		$this->expectExceptionMessage('Array sizes are inconsistent');
```

Reply:
I'm not actually sure if that is the message bc my PHP 8 system is down right now but we will try it out :)